src/record/field.rs

```diff
@@ -873,6 +873,13 @@ fn read_string_of_len<T: Read>(source: &mut T, len: u8) -> Result<String, std::i
     let mut bytes = Vec::<u8>::new();
     bytes.resize(len as usize, 0u8);
     source.read_exact(&mut bytes)?;
+    // Trims the null bytes: string cannot be properly trimmed otherwise
+    let trimmed_bytes = bytes.split(|b| b == &b'\0').next();
+    let bytes = if let Some(a) = trimmed_bytes {
```

Comment:
tmontaigu now

I have a few comments here:

    I think using a match would make the code clearer and shorter
    We are doing an unneeded allocation + copy by using Vec::from

I would suggest

    let trimmed_bytes = match bytes.split(|b| *b == 0).next() {
        Some(trimmed_bytes) => trimmed_bytes,
        None => &bytes
    };
    Ok(String::from_utf8_lossy(trimmed_bytes).into_owned());

Reply:
Thanks for the feedback, I made the adjusments suggested !